https://sourceforge.net/p/mediatomb/bugs/90/

#90 Unable to build with new FFMPEG API
Status: open
Owner: nobody
Labels: Building From Source (19)
Priority: 8
Updated: 2011-04-30
Created: 2011-04-21
Creator: Kyle
Private: No

The latest FFMPEG has removed the depreciated api that src/metadata/ffmpeg_handler.cc was using to get metadata:

CODEC_TYPE_VIDEO is now AVMEDIA_TYPE_VIDEO
CODEC_TYPE_AUDIO is now AVMEDIA_TYPE_AUDIO


https://launchpadlibrarian.net/71985647/libav_0.7_support.patch

=== modified file 'src/metadata/ffmpeg_handler.cc'
--- old/src/metadata/ffmpeg_handler.cc	2010-08-25 17:07:03 +0000
+++ new/src/metadata/ffmpeg_handler.cc	2011-05-19 04:33:32 +0000
@@ -89,6 +89,33 @@
 
 	Ref<StringConverter> sc = StringConverter::m2i();
     
+	/* Tabs are 4 characters here */
+	typedef struct {const char *avname; metadata_fields_t field;} mapping_t;
+	static const mapping_t mapping[] =
+	{
+		{"title",   M_TITLE},
+		{"artist",  M_ARTIST},
+		{"album",   M_ALBUM},
+		{"date",    M_DATE},
+		{"genre",   M_GENRE},
+		{"comment", M_DESCRIPTION},
+		{"track",   M_TRACKNUMBER},
+		{NULL,      M_MAX},
+	};
+
+	if (!pFormatCtx->metadata)
+		return;
+	for (const mapping_t *m = mapping; m->avname != NULL; m++)
+	{
+		AVMetadataTag *tag = NULL;
+		tag = av_metadata_get(pFormatCtx->metadata, m->avname, NULL, 0);
+		if (tag && tag->value && tag->value[0])
+		{
+			log_debug("Added metadata %s: %s\n", m->avname, tag->value);
+			item->setMetadata(MT_KEYS[m->field].upnp, sc->convert(tag->value));
+		}
+	}
+	/* Old algorithm (doesn't work with libav >= 0.7)
 	if (strlen(pFormatCtx->title) > 0) 
     {
 	    log_debug("Added metadata title: %s\n", pFormatCtx->title);
@@ -131,6 +158,7 @@
         item->setMetadata(MT_KEYS[M_TRACKNUMBER].upnp, 
                           sc->convert(String::from(pFormatCtx->track)));
 	}
+	*/
 }
 
 // ffmpeg library calls
@@ -178,7 +206,7 @@
 	for(i=0; i<pFormatCtx->nb_streams; i++) 
     {
 		AVStream *st = pFormatCtx->streams[i];
-		if((st != NULL) && (videoset == false) && (st->codec->codec_type == CODEC_TYPE_VIDEO))
+		if((st != NULL) && (videoset == false) && (st->codec->codec_type == AVMEDIA_TYPE_VIDEO))
         {
             if (st->codec->codec_tag > 0)
             {
@@ -209,7 +237,7 @@
                 *y = st->codec->height;
 			}
 		} 
-		if(st->codec->codec_type == CODEC_TYPE_AUDIO) 
+		if(st->codec->codec_type == AVMEDIA_TYPE_AUDIO) 
         {
 			// Increase number of audiochannels
 			audioch++;

